test/scenarios.py

```diff
@@ -6730,6 +6730,84 @@
             ),
         },
     ),
+    Scenario(
+        "scenario29-13",
```

Comment:
This is a good test. It also highlights that:
- the sender is oblivious that a file was removed from the filesystem
- probably no failed state is inserted into the database on the sender side

Reply:
Yeah, I'm not sure how to fix that or is it worth fixing

Reply:
I've approved the PR still as I agree that this might be minor(who really deletes the files when sending), however, it shouldn't be too hard to go via failed file transfer way?

Reply:
It actually is.
So the sending of chunks is initiated by the receiver and if it finds a temporary it requests for the checksum.
The checksum fails on the sender and is communicated back to the receiver.
The receiver emits an error event and what's crucial **does not issue a download message to the sender**.

So in the end the receiver knows that the file transfer failed but the sender doesn't because it really failed before he was made aware of the upload at all.

As I mentioned it should behave differently but I honestly don't know how involving fixing it would be. Probably would require introducing a new protocol version so that the sender can relate checksum requests with a particular transfer which we will do eventually anyway.
So I would leave it as a ticket for 5.1

Reply:
I think this is a separate issue. I meant that if the sender initiates the transfer, goes offline, deletes the file and comes online he should expect FileFailed state in his database

Reply:
I mean it's basically the same because we insert state into DB based on events emitted